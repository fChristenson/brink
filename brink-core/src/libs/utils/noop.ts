import { IOnSubmit } from "../../store/context";

export const noop = () => {
  return undefined;
};

export const submitNoop: IOnSubmit = (_: any) => {
  return false;
};
