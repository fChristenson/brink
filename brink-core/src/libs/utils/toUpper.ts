export const toUpper = (val: any): string | undefined => {
  try {
    return val.toUpperCase();
  } catch (e) {
    return undefined;
  }
};
