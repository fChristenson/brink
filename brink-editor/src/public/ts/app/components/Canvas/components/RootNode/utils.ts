import { IRootNode } from "./RootNode";
import { IConnection } from "./Connection";

export const getTransform = (x: number, y: number) =>
  `translate(${x}px, ${y}px)`;

export const hasConnection = (rootNode: IRootNode) => (c: IConnection) =>
  c.from === rootNode.id || c.to === rootNode.id;

export const hasOutgoingConnection = (rootNode: IRootNode) => (
  c: IConnection
) => c.from === rootNode.id;

export const capitalize = (str: string) => {
  if (str.length <= 1) return str;
  const [first, ...rest] = str.split("");

  return first.toUpperCase() + rest.join("");
};
