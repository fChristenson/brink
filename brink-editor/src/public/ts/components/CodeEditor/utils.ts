import { xml2json } from "xml-js";
import { isValidTag } from "brink-core";

export const parseXml = (xml: string) => {
  const xmlStr = xml2json(xml);
  return JSON.parse(xmlStr);
};

export const isValidXml = (xml: any): boolean => {
  if (!xml) return false;
  if (xml.type === "text") return true;

  const result = [];

  if (xml.type === "element") {
    result.push(isValidTag(xml.name));
  }

  for (const element of xml.elements || []) {
    result.push(isValidXml(element));
  }

  return result.every(bool => bool === true);
};
