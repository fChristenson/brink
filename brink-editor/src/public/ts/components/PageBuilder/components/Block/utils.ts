export const highlightBlocks = (highlight: boolean) => {
  document.body.style.setProperty(
    "--highlight-containers-opacity",
    highlight ? "1" : "0"
  );
  document.body.style.setProperty(
    "--highlight-containers-pointer",
    highlight ? "pointer" : "initial"
  );
};
