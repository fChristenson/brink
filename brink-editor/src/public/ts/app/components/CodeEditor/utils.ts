import { xml2json } from "xml-js";
import { isValidTag } from "brink-core";
import { IRootNode } from "../Canvas/components/RootNode/RootNode";

export const findRootNode = (rootNodes: IRootNode[], id: string): IRootNode => {
  const rootNode = rootNodes.find(n => n.id === id);
  if (!rootNode) throw new Error(`Could not find ${id}`);

  return rootNode;
};

export const parseXml = (xml: string) => {
  const xmlStr = xml2json(xml);
  return JSON.parse(xmlStr);
};

export const isValidXml = (xml: any): boolean => {
  if (!xml) return false;
  if (xml.type === "text") return true;

  const result = [];

  if (xml.type === "element") {
    result.push(isValidTag(xml.name));
  }

  for (const element of xml.elements || []) {
    result.push(isValidXml(element));
  }

  return result.every(bool => bool === true);
};
