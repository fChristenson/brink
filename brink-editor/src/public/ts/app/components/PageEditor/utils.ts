import { contentClassName } from "../Content";
import domtoimage from "dom-to-image";
import { IRootNode } from "../Canvas/components/RootNode/RootNode";

export const takeScreenshot = (
  rootNode: IRootNode,
  onScreenShot: (rootNode: IRootNode, dataUrl: string) => void
) => {
  const node = document.querySelector(`.${contentClassName} > main`);
  if (!node) throw new Error(`Could not find ${contentClassName} > main node`);

  return domtoimage.toPng(node).then(imgData => {
    onScreenShot(rootNode, imgData);
  });
};
