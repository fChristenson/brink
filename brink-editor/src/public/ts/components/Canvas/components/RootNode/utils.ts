import { IConnection, IRootNode } from "./RootNode";

export const getTransform = (x: number, y: number) =>
  `translate(${x}px, ${y}px)`;

export const hasConnection = (rootNode: IRootNode) => (c: IConnection) =>
  c.from.id === rootNode.id || c.to.id === rootNode.id;
