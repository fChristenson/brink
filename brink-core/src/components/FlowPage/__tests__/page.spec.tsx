import React from "react";
import { PageComponent as Page, styles } from "..";
import { childrenTests } from "../../../libs/tests/childrenTest";
import { baseTests } from "../../../libs/tests/baseTests";
import { submitNoop } from "../../../libs/utils/noop";

describe("Page tests", () => {
  baseTests(<Page onSubmit={submitNoop} />, styles);
  childrenTests(<Page onSubmit={submitNoop} />);
});
