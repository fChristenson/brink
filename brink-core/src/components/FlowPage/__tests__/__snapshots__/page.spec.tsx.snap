// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page tests Base test should assert that the component matches the snapshot 1`] = `"<main style=\\"padding:10px 10px 18px;background-color:white\\"><form style=\\"margin:0\\"></form></main>"`;

exports[`Page tests Base test should assert that the component matches the snapshot 2`] = `
Object {
  "backgroundColor": "white",
  "padding": "10px 10px 18px",
}
`;
